Vrr single contig reference aligner

vruano opened the pull request.

vruano commented on src/main/java/org/broadinstitute/hellbender/tools/spark/sv/utils/SingleReferenceSequenceAligner.java:
FYI I will rename The class to SingleContigReferenceAligner and make it final.

vruano commented:
It can easely be extended to multiple sequences... it just happened that I didn't need it personally. 

I took a quick pick to your branch... do you really care about the contig descriptions?.... I would say that this "aligner" class should not be responsible of compose the multi sequence fasta file but rather accept one as a constructor argument and the construction of the fasta is delegated back to the invoker code; in this new more general aligner the current single contig could be implemented as a public static method call.

vruano commented:
In addition just to clarify that the "contig" word reference to the sequences in the given reference not the number of contigs that you might align to it, so single-contig-reference-aligner means an aligner that will align several sequences (read, contigs, random-stuff) vs a single sequence/contig reference. 

I guess I shall change contig for chromosome or sequence to make it more clear.

SHuang-Broad commented on src/test/java/org/broadinstitute/hellbender/utils/RandomDNA.java:
```diff
@@ -20,11 +18,44 @@
+ *     not the sequence of calls to different base generating methods.
+ *
+ *     For example a single call asking for 132 bases using {@link #nextBases(int)} would generate the same sequence
+ *     as calling 132 times {@link #nextBase()}.
```
if initialized with the same seed

vruano commented:
@SHuang-Broad 
I have addressed comments and changed the SSRAligner interface to make it more general in terms of the type of inputs (base sequence providers) and the type of outputs that it will generate. 

Also I added and example as to how to address your SVFastqUtil.FastqRead to SAMRecord realigner may look like but is untested. perhaps you can take on that commit and add the test, 

does not need to be part of this PR push though.
